src/ui/public/index_patterns/_calculate_indices.js

```diff
@@ -2,6 +2,12 @@ define(function (require) {
   const _ = require('lodash');
   const moment = require('moment');
 
+  // gets parsed value if given arg is a moment object
+  function timeValue(val) {
+    const hasValueOfFn = _.isFunction(_.get(val, 'valueOf'));
```

Comment:
isn't this a complex version of `start && start.valueOf()`. Everything except undefined and null has a valueOf method.